### ISSUE TITLE
DatePicker on External Input Element

### ISSUE THREAD

**Issue by nsisodiya** [Hacker0x01/react-datepicker#340]
This should also work.

``` js
<DatePicker
    selected={this.state.startDate}
    onChange={this.handleChange} 
    className='red-border' >
     <input type='text' placeholder="Enter Date">
</DatePicker>
```

Or other way.
For me DatePicker is  a not a control. Date Picker should be a `Effect` on a control.
for example, My desired input can be anywhere or any library can generate it. 

We can have a optionally. as User may want to send external input element or not.

**Comment by rafeememon** [comment]
I like this idea; here are a couple things I thought of

* I think this should probably be exposed as a separate component, possibly `DatePickerWrapper` to separate logic (currently the `DatePicker` and the `DateInput` are pretty tightly coupled).  As a bonus, this could be a mixin, or as an extra bonus, an annotation??
* Open state would need to be entirely managed by the user since
  * we probably don't want to attach random callbacks to the child component
  * most elements don't have focus/blur state

**Comment by i4got10** [comment]
@rafeememon Probably there is no need to replace whole ``DateInput``. That is really need is to replace [native input inside](https://github.com/Hacker0x01/react-datepicker/blob/master/src/date_input.jsx#L86). Looks like its not so hard

**Comment by gpoole** [comment]
@nsisodiya this may be mostly resolved by #551, is that right?

**Comment by martijnrusschen** [comment]
Yep, looks like it. Closing due inactivity.
